tests/test_dir_watcher.py

```diff
@@ -0,0 +1,257 @@
+"""dir_watcher tests"""
+
+import os
+from typing import Callable, TYPE_CHECKING
+from unittest.mock import Mock, call
+
+import pytest
+import py.path
```

Comment:
why not use `pathlib`... i think this is the recommended package for new code

Reply:
It's not actually entirely in our control! Pytest provides a `tmpdir` fixture, and its type is `py.path.local`.

...eh, but I can refactor my `dir_watcher` fixture to just use `tempfile.TemporaryDirectory`.

Reply:
I asked just because of this : https://py.readthedocs.io/en/latest/path.html#py-path